docs: add permissions information (#3116)
garden-io/garden: docs/guides/remote-kubernetes.md

You'll also need to point one or more DNS entries to your cluster, and configure a TLS certificate for the hostnames
you will expose for ingress.

Templating the ingress to the application enables you to have DNS entries for every developer's namespace.  

First, you will make DNS CNAME entry that points to the load balancer in front of your cluster. We recommend setting a wildcard in front of the proper record, e.g. *.<environment>.<your company>.com.

In your project.garden.yaml file, templatize the hostnames for each environment. You can use some of the built-in templating functions garden provides such as the local username.

```yml
kind: Project
name: my-project
defaultEnvironment: dev
environments:
  - name: dev
    defaultNamespace: ${var.username}-my-project
    variables:
        hostname: ${local.username}-my-project.dev.mydomain.com
 - name: preview
    defaultNamespace: preview-${local.env.PR_NUMBER}
    variables:
        hostname: ${local.env.PR_NUMBER}-my-project.preview.mydomain.com
providers:
  - name: kubernetes
    environments: [dev, preview]
    namespace: ${environment.namespace}
    buildMode: cluster-buildkit
    defaultHostname: ${var.hostname}
```

> If you would like to manage TLS for development environments, we recommend using an application load balancer and your cloud provider's certificate management service. You can find the documentation for (AWS here)[https://aws.amazon.com/premiumsupport/knowledge-center/associate-acm-certificate-alb-nlb/] and for (GCP here)[https://cloud.google.com/load-balancing/docs/ssl-certificates/google-managed-certs].

Reviewer: ```suggestion
If you would like to manage TLS for development environments, we recommend using your cloud provider's certificate management service in combination with a load balancer. You can find the documentation for (AWS here)[https://aws.amazon.com/premiumsupport/knowledge-center/associate-acm-certificate-alb-nlb/] and for (GCP here)[https://cloud.google.com/load-balancing/docs/ssl-certificates/google-managed-certs].
```
This would also work with other types of load balancers like network loadbalancers, so removing the application here.
I have been thinking once more about this section and I think it requires its own page with detailed instructions for the different possibilities and cloud providers. The sentence above is actually not enough to really help someone new to Kubernetes implement load balancing with TLS. 
I think we can merge this for now but I would really like to create a proper guide around it that covers different cloud providers and options in more detail because something that works for EKS in AWS might not work for GKE Autopilot clusters in GCP. cc @worldofgeese

Author: Thank you! I've made an Asana ticket for this 👍🏻
